enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,725 @@
+  * `message` (optional, [string][json-string]), with a human-oriented message describing the blocking reason, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+  * `blockedClusters` (optional, [object][json-object]).
+    If any `blockedClusters` property matches a given cluster, the edge is not recommended for that cluster, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+    If `blockedClusters` is unset, the edge is blocked for all clusters.
+    * `promql` (optional, [string][json-string]), defining the subset of affected clusters, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+
+### Enhanced ClusterVersion representation
+
+[The ClusterVersion `status`][api-cluster-version-status] will be extended with a new `conditionalUpdates` property:
+
+```go
+// conditionalUpdates contains the list of updates that may be
+// conditionally recommended for this cluster. Consumers interested
+// in the set of updates that are actually recommended for this
+// cluster should use availableUpdates. This list may be empty if no
+// updates are recommended, if the update service is unavailable, or
+// if an empty or invalid channel has been specified.
+// +optional
+conditionalUpdates []ConditionalUpdate `json:"conditionalUpdates,omitempty"`
+```
+
+The `availableUpdates` documentation will be adjusted to read:
+
+```go
+// availableUpdates contains the subset of conditionalUpdates that
+// apply to this cluster.  Updates which appear in conditionalUpdates
+// but not in availableUpdates may expose this cluster to known
+// issues.
+```
+
+The new ConditionalUpdate type will have the following schema:
+
+```go
+// ConditionalUpdate represents an update which is recommended to some
+// clusters on the version the current cluster is reconciling, but which
+// may not be recommended for the current cluster.
+// +k8s:deepcopy-gen=true
+type ConditionalUpdate struct {
+	// release is the target of the update.
+	// +required
+	release Release `json:"release"`
+
+	// blockedClusters represents the subset of clusters for which the conditional update is not recommended.
+	// +required
+	BlockedClusters []ConditionalUpdateBlocker `json:"blockedClusters"`
+
+	// conditions represents the observations of the conditional update's
+	// current status. Known types are:
+	// * Recommended, for whether the update is recommended for the current cluster.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+}
+```
+
+The new ConditionalUpdateBlocker type will have the following schema:
+
+```go
+// ConditionalUpdateBlocker represents a reason and cluster-set for
+// blocking a conditional update.
+// +k8s:deepcopy-gen=true
+type ConditionalUpdateBlocker struct {
```

Comment:
RiskAssessment?  RiskCriteria?

Reply:
Let's fold this discussion into [this existing thread][1] about the consuming property name.  It doesn't seem likely that we will want a value-type structure name that wildly diverges from the consuming property name.

[1]: https://github.com/openshift/enhancements/pull/821#discussion_r698763302